allennlp/models/model.py

```diff
@@ -147,7 +182,18 @@ def get_metrics(self, reset: bool = False) -> Dict[str, float]:
     @classmethod
     def from_params(cls, vocab: Vocabulary, params: Params) -> 'Model':
         choice = params.pop_choice("type", cls.list_available())
-        return cls.by_name(choice).from_params(vocab, params)
+        regularizer_params = params.pop("regularizer", None)
+        initializer_params = params.pop("initializer", None)
+
+        model = cls.by_name(choice).from_params(vocab, params)
+
+        if regularizer_params is not None:
+            model.add_regularizer(RegularizerApplicator.from_params(regularizer_params))
+        if initializer_params is not None:
+            model.add_initializer(InitializerApplicator.from_params(initializer_params))
+            model.initialize()
```

Comment:
I don't think this goes here.  The trouble is there's no good spot to put it.  I want it to be in the constructor, but it has to be called _after_ all submodules have been assigned to `self`, while pytorch requires that the super constructor be called _before_ any submodules have been assigned to `self`...

Reply:
Passing the initializer to the super class constructor doesn't solve the issue, either, because the modules wouldn't have been assigned...

Reply:
we could put it in the trainer and call it if you're not starting from a checkpoint